lib/portlayer/storage/vsphere/datastore.go

```diff
@@ -301,3 +303,36 @@ func (d *Datastore) mkRootDir(ctx context.Context, rootdir string) error {
 func (d *Datastore) rootDir() string {
 	return strings.SplitN(d.RootURL, " ", 2)[1]
 }
+
+// Parse the datastore format ([datastore1] /path/to/thing) to groups.
+var datastoreFormat = regexp.MustCompile(`\[([\w\d]*)\]\ (.*)?`)
+
+// Converts `[datastore] /path` to URL
+func DatastoreToURL(ds string) (*url.URL, error) {
+	matches := datastoreFormat.FindAllStringSubmatch(ds, -1)
+
+	if len(matches) != 1 || len(matches[0]) != 3 {
+		return nil, fmt.Errorf("invalid datastore format: %s", ds)
+	}
+
+	datastoreName := matches[0][1]
```

Comment:
Also, spaces allowed in datastore path? Can you omit it altogether? Your regex allows those cases too.